ibis/pandas/execution/generic.py

```diff
@@ -590,99 +609,11 @@ def execute_null_if_zero_series(op, data, **kwargs):
     return data.where(data != 0, np.nan)
 
 
-@execute_node.register(ops.StringLength, pd.Series)
-def execute_string_length_series(op, data, **kwargs):
-    return data.str.len()
-
-
 @execute_node.register(
-    ops.Substring,
-    pd.Series,
-    (pd.Series,) + integer_types,
-    (pd.Series,) + integer_types
+    ops.StringSplit, pd.Series, (pd.Series,) + six.string_types
 )
-def execute_string_substring(op, data, start, length, **kwargs):
-    return data.str[start:start + length]
-
-
-@execute_node.register(ops.Strip, pd.Series)
-def execute_string_strip(op, data, **kwargs):
-    return data.str.strip()
-
-
-@execute_node.register(ops.LStrip, pd.Series)
-def execute_string_lstrip(op, data, **kwargs):
-    return data.str.lstrip()
-
-
-@execute_node.register(ops.RStrip, pd.Series)
-def execute_string_rstrip(op, data, **kwargs):
-    return data.str.rstrip()
-
-
-@execute_node.register(
-    ops.LPad,
-    pd.Series,
-    (pd.Series,) + integer_types,
-    (pd.Series,) + six.string_types
-)
-def execute_string_lpad(op, data, length, pad, **kwargs):
-    return data.str.pad(length, side='left', fillchar=pad)
-
-
-@execute_node.register(
-    ops.RPad,
-    pd.Series,
-    (pd.Series,) + integer_types,
-    (pd.Series,) + six.string_types
-)
-def execute_string_rpad(op, data, length, pad, **kwargs):
-    return data.str.pad(length, side='right', fillchar=pad)
-
-
-@execute_node.register(ops.Reverse, pd.Series)
-def execute_string_reverse(op, data, **kwargs):
-    return data.str[::-1]
-
-
-@execute_node.register(ops.Lowercase, pd.Series)
-def execute_string_lower(op, data, **kwargs):
-    return data.str.lower()
-
-
-@execute_node.register(ops.Uppercase, pd.Series)
-def execute_string_upper(op, data, **kwargs):
-    return data.str.upper()
-
-
-@execute_node.register(ops.Capitalize, pd.Series)
-def execute_string_capitalize(op, data, **kwargs):
-    return data.str.capitalize()
-
-
-@execute_node.register(ops.Repeat, pd.Series, (pd.Series,) + integer_types)
-def execute_string_repeat(op, data, times, **kwargs):
-    return data.str.repeat(times)
-
-
-@execute_node.register(
-    ops.StringFind,
-    pd.Series,
-    (pd.Series,) + six.string_types,
-    (pd.Series, type(None)) + integer_types,
-    (pd.Series, type(None)) + integer_types,
-)
-def execute_string_contains(op, data, needle, start, end, **kwargs):
-    return data.str.find(needle, start, end)
-
-
-@execute_node.register(
-    ops.StringSQLLike,
-    pd.Series,
-    (pd.Series,) + six.string_types,
-)
-def execute_string_like(op, data, pattern, **kwargs):
-    return data.str.contains(pattern, regex=True)
+def execute_string_split(op, data, delimiter, **kwargs):
```

Comment:
No, `kwargs` is always non-empty because we pass `context` around everywhere. All aggregations use the context argument. We'd have to filter out unused arguments using `inspect` *before* calling `execute_node` which seems like a lot of code and hassle for not much benefit.

I suppose one nice thing about doing that is that you wouldn't have to include `**kwargs` in every signature. When I wrote this I didn't think that would be a huge inconvenience.